### ISSUE TITLE
Pipewire-aes67 module compatibility

### ISSUE THREAD

**Issue by dewiweb** [bondagit/aes67-linux-daemon#148]
Is someone already test aes67-linux-daemon in convergence with pipewire-aes67 module? I'd like to understand the complementarity of those two projects. At which level each implementation works?

**Comment by Play-AV** [comment]
^^ I haven't played with Pipewire's AES67 stuff yet. Have you been using it at all? Very curious.

**Comment by sh7dm** [comment]
I develop PipeWire AES67 support. It's unrelated to this project and fully independent from the Merging kernel-mode driver

**Comment by bondagit** [comment]
> I develop PipeWire AES67 support. It's unrelated to this project and fully independent from the Merging kernel-mode driver

Just for curiosity Is PTP slave clock sync implemented on PipeWire AESS67 module ?

**Comment by sh7dm** [comment]
No, we currently rely on ptp4l (typically) syncing the clock. Reimplementing all that code doesn't seem feasible, also because PipeWire runs as a user unit and cannot access port 319/320 and adjust clocks. We use PHC (/dev/ptp*) clocks to adjust both stream position and rate of activation (this also makes applications using the AES67 sink follow the GM clock).

Do you know if ptp4l has an API to get the current GM identity (MAC)? We currently do this by using `pmc` and then manually filling out the field (or copying it from other devices' SDP data)

**Comment by cloudcastsystemsau** [comment]
@dsseng how do you change the rate of sending rtp packets? Do you have 1ms timer, how does ptp4l influence the rate of sending packets?

**Comment by dsseng** [comment]
They're sent out in accordance to driver clock, which is connected to the PHC device, which is in turn adjusted by ptp4l to match network clock

**Comment by JarkkoRx** [comment]
I actually have been testing the functionality of aes67-linux-daemon and pipewire-aes67 simultaneously (obviously on different systems), and, nothing surprising here, interoperability works as expected. I actually do hope that @bondagit and pipewire-aes67 team would collaborate in some way, just to make one perfect open source solution for AES67. Let's hope Windows would also get something from this front :)

**Comment by dsseng** [comment]
Happy to know that! Well, having AES67 now might be a good reason to poke around getting PW on other platforms if possible

**Comment by Play-AV** [comment]
> I actually have been testing the functionality of aes67-linux-daemon and pipewire-aes67 simultaneously (obviously on different systems), and, nothing surprising here, interoperability works as expected. I actually do hope that @bondagit and pipewire-aes67 team would collaborate in some way, just to make one perfect open source solution for AES67. Let's hope Windows would also get something from this front :)

Windows works well for me. 
You can do Audinate's Dante VSC or roll your own.

**Comment by Matzeee** [comment]
> > I actually have been testing the functionality of aes67-linux-daemon and pipewire-aes67 simultaneously (obviously on different systems), and, nothing surprising here, interoperability works as expected. I actually do hope that @bondagit and pipewire-aes67 team would collaborate in some way, just to make one perfect open source solution for AES67. Let's hope Windows would also get something from this front :)
> 
> Windows works well for me. 
> You can run Audinate's Dante VSC or roll your own. 
> 
> 

How do you use Aes67 on Windows?
Afaik, Dantes VSC does not support Aes67 on its own?

What do you mean by "roll your own"?
